spacewalk-hostname-rename - basic info (#263)
uyuni-project/uyuni-docs: modules/administration/pages/hostname-rename.adoc

[[hostname-rename]]
= Rename Hostname

With the [command]``spacewalk-hostname-rename`` script it is possible to rename the hostname of the {productname} Server.
The [command]``spacewalk-hostname-rename`` script is part of the [package]``spacewalk-utils`` package.

The only mandatory parameter for the script is the newly configured IP address of the {productname} Server.

> It is the best to proceed as follows: 

Reviewer: Make it mandatory. I don't know what happens if the user do it otherwise, but maybe things will not work :-)

> It is the best to proceed as follows: 

Reviewer: uyuni-master-srv.mgr.suse.de

Reviewer: I don't understand why I added this. The original comment was something like:

> I would not say, this, but just "Follow this procedure". For me "It is best" means "I can do follow another procedure if I think it's better" :-D

. Adjust the hostname in advance.

> It is important to configure it locally, and for DNS resolution (including reverse resolution).

Reviewer: I would say: "And at the DNS severs"

Author: ok, will do.

> . Load the new configuration.

Reviewer: This is not about new configuration, but about the server using the new hostname. Otherwise the script complains.

Author: ok, I changed it for clarity.

> If in doubt whether the new configuration is running, reboot the server.

Reviewer: Wouldn't it be easier if we just say "Reboot the server" without conditionals?

Author: if done right, rebooting is not needed.  I do not like recommending a reboot if it is not strictly necessary.  Nevertheless I'll change the text.  If you (one of you) still disagree, I can simplify it even more.

. Run the script [command]``spacewalk-hostname-rename`` script with the public IP address of the server.

> . Some other actions are needed to make your landscape aware of the new hostname and IP address.

Reviewer: I strongly recommend you check at the galaxy-devel mailing lists what other actions are required, so we can list them here.

As I sad at the card:

> NOTE: I only did basic testing, someone else with more experience with the script should help doc team.

Author: ok

[[hostname-rename]]
= Rename Hostname

With the [command]``spacewalk-hostname-rename`` script it is possible to rename the hostname of the {productname} Server.
The [command]``spacewalk-hostname-rename`` script is part of the [package]``spacewalk-utils`` package.

The only mandatory parameter for the script is the newly configured IP address of the {productname} Server.

.Procedure: Renaming Hostname

> . Adjust the hostname in advance.

Reviewer 2: There's something really jarring about this. If the procedure is "renaming hostname" I don't expect step one to be "you should already have renamed the hostname".
I'm not sure what to do about it, except for describing this procedure in more detail beforehand.

Reviewer: As strange as it is, the script only takes care of renaming things at SUSE Manager/Uyuni Server level, but not on DNS/locall resolution level.

Reviewer 2: Yeah, I just think we need to explain that before we launch into the procedure, to avoid confusion.

Reviewer 3: Maybe the confusion comes from the fact that "renaming hostname" is a somewhat generic term. What the "rename-hostname" script is really doing is entering the information of a changed hostname into the database and internal structures of SUSE Manager. So the hostname rename on system level is something the admin has to do on his own like with any regular system. Our script will ensure that SUSE Manager is aware that the host it runs on has been renamed.

Author: ok, thanks!  I hope I got it right.

It is important to configure it locally, and at the DNS server for DNS resolution.
You must also provide configuration settings for reverse name resolution.
. Restart your network configuration to make the {productname} server use the new hostname.

> If the server does not use the new name, the renaming script will complain.

Reviewer 2: ```suggestion
If the server does not use the new name, the script will not complete.
```

[[hostname-rename]]
= Rename {productname} Server

It is possible to rename a {productname} Server.

> You first must change the network settings of the server locally and remotely.

Reviewer 2: ```suggestion
Begin by changing the network settings of the server locally and remotely.
```

> Then the [command]``spacewalk-hostname-rename`` script will the settings in the {productname} PostgreSQL database and the internal structures of {productname} accordingly.

Reviewer 2: ```suggestion
Then use the [command]``spacewalk-hostname-rename`` script to update the settings in the PostgreSQL database and the internal structures of {productname}.
```

The [command]``spacewalk-hostname-rename`` script is part of the [package]``spacewalk-utils`` package.

The only mandatory parameter for the script is the newly configured IP address of the {productname} Server.

.Procedure: Renaming {productname} Server
. Change the network settings of the server on the system level locally and remotely at the DNS server.

> This part of the procedure is similar to the actions to take if you want to rename any other regular system.

Reviewer 2: ```suggestion
This is done in the same way as renaming any other system.
```

+
[NOTE]
====
You must also provide configuration settings for reverse name resolution.

> ====

Reviewer 2: I'm increasingly disliking admonitions in the middle of procedures, because our formatting (or lack thereof :confused: ) doesn't handle it well. Since it's only a note, can we just write it inline instead?

Author: ok

> . Restart your network configuration to make the {productname} server use the new hostname.
> If the server does not use the new name, the script will not complete.
> After rebooting the server will use the new name.

Reviewer 2: ```suggestion
. Restart your network configuration to make the {productname} Server use the new hostname.
. Reboot the {productname} Server to ensure the hostname has changed.
If the server is not using the new name, the script will fail.
```

. Run the script [command]``spacewalk-hostname-rename`` script with the public IP address of the server.

> . Some other actions are needed to make your landscape aware of the new hostname and IP address.

Reviewer 2: Landscape? :thinking: Do you mean environment?

Reviewer 2: I also don't like the vague "some other actions" but that might be the best we can do under the circumstances.

Reviewer: I'd give some examples: AFAIK traditional and salt clients MUST be reconfigured as otherwise they will still point to the old hostname. Not sure if "reconfiguring" is the word, or "re-onboarding". Someone with experience should know.

Reviewer 4: I think re-configure is the right word. Think about salt minion config file. You specify

```
master: FQDN.OF.MASTER
```

If you change that hostname, the minion needs to be re-configured.
The traditional client has /etc/sysconfig/rhn/up2date configfile which must be changed.

> Change all Salt clients to point to the new master, and with a re-activation key re-register the traditional clients (if there are any).

Reviewer 2: Can probably drop some xrefs in here.

Reviewer: Ah, wait here it is!

Then I am not sure what "other actions" could mean. An expert should advice :-)

Reviewer 4: I am not sure if you need to re-onboard / re-register the clients. Would be good to try this out.

Reviewer 3: Btw, I think if there are proxies that are being used for autoinstallations, it might be needed to reconfigure tftp-sync-recv (do not remember the right name from head, but I think you know what I mean) on the affected proxies as well.

[[hostname-rename]]
= Rename {productname} Server

It is possible to rename a {productname} Server.
Begin by changing the network settings of the server locally and remotely.
Then use the [command]``spacewalk-hostname-rename`` script to update the settings in the PostgreSQL database and the internal structures of {productname}.

The [command]``spacewalk-hostname-rename`` script is part of the [package]``spacewalk-utils`` package.

The only mandatory parameter for the script is the newly configured IP address of the {productname} Server.

.Procedure: Renaming {productname} Server
. Change the network settings of the server on the system level locally and remotely at the DNS server.
Provide configuration settings for reverse name resolution, too.
Changing network settings is done in the same way as with renaming any other system.
. Restart your network configuration to make the {productname} Server uses the new hostname.

> . Reboot the {productname} Server to ensure the hostname has changed.

Reviewer: If you reboot the server, you will get the new network configuration as well. Can't we just make this a single step?

Author: ok, I simplified it.

> If the server is not using the new name, the script will fail.

Reviewer: Shouldn't this a note?

Author: Within procedure, we no longer want admonitions such as formal notes.  Maybe, we can fall back to fake notes, starting with `NOTE:` on a new line.

Reviewer 2: alternatively, you could do it as a proper admonition at the end of the procedure? @jcayouette this is probably an important thing to consider when you are talking to Naman about block styling, btw.

Author: for  a simple NOTE it is probably ok to have it at the end of the procedure.
